src/ucp/core/ucp_am.c

```diff
@@ -1109,11 +1116,13 @@ ucp_am_handle_unfinished(ucp_worker_h worker, ucp_recv_desc_t *first_rdesc,
     ucs_list_del(&first_rdesc->am_first.list);
 
     first_hdr = (ucp_am_first_hdr_t*)(first_rdesc + 1);
-    reply_ep  = ucp_am_hdr_reply_ep(worker, first_hdr->super.super.flags,
-                                    first_hdr->super.ep_id);
-    status    = ucp_am_invoke_cb(worker, &first_hdr->super.super,
-                                 sizeof(*first_hdr), first_hdr->total_size,
-                                 reply_ep, 1);
+
+    ucp_am_hdr_reply_ep(worker, first_hdr->super.super.flags,
+                        first_hdr->super.ep_id, &reply_ep);
+
+    status = ucp_am_invoke_cb(worker, &first_hdr->super.super,
```

Comment:
can we remove `if (reply_ep != NULL) {` in ucp_am_invoke_cb by passing reply_ep and recv_attr from here?